File: code-example-tests/javascript/driver/examples/get-started/get-started.js
//	:replace-start: {
//	  "terms":
//	  {
//	  "process.env.CONNECTION_STRING": "\"<connection string URI>\"",
//    "export ": ""
//	  }
//	}
import { MongoClient } from 'mongodb';

// :snippet-start: example
// :uncomment-start:
//const { MongoClient } = require("mongodb");
// :uncomment-end:

export async function runGetStarted() {
  // Replace the uri string with your connection string
  const uri = process.env.CONNECTION_STRING;
  const client = new MongoClient(uri);

  try {
    const database = client.db('sample_mflix');
    const movies = database.collection('movies');

    // Queries for a movie that has a title value of 'Back to the Future'
    const query = { title: 'Back to the Future' };
    const movie = await movies.findOne(query);
    // :uncomment-start:
    //console.log(movie);
    // :uncomment-end:
    return movie; // :remove:
  } finally {
    await client.close();
  }
}
// :uncomment-start:
//runGetStarted().catch(console.dir);
// :uncomment-end:
// :snippet-end:
// :replace-end:

File: code-example-tests/python/pymongo/examples/get_started/get_started.py
#   :replace-start: {
#     "terms": {
#       "CONNECTION_STRING": "\"<connection string URI>\""
#     }
#   }

def get_started(CONNECTION_STRING):
    # :snippet-start: example
    from pymongo import MongoClient
    import json

    uri = CONNECTION_STRING
    client = MongoClient(uri)

    try:
        database = client.get_database("sample_mflix")
        movies = database.get_collection("movies")

        # Queries for a movie that has the title 'Back to the Future'
        query = { "title": "Back to the Future" }
        movie = movies.find_one(query)

        print(json.dumps(movie, indent=4, default=str))

        client.close()
        return movie  # :remove:

    except Exception as e:
        raise Exception("Unable to find the document due to the following error: ", e)

# :snippet-end:
# :replace-end:

